Placeeo/Controller/LocationManager.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  LocationManager.swift
+//  Placeeo
+//
+//  Created by Pasha Berger on 03.10.2021.
+//
+
+import Foundation
+import CoreLocation
+
+class LocationManager: NSObject, CLLocationManagerDelegate {
+    static let shared = LocationManager()
+    
+    let manager = CLLocationManager()
+    
+    var completion: ((CLLocation) -> Void)?
+    
+    public func getUserLocation(completion: @escaping ((CLLocation) -> Void)) {
+        self.completion = completion
+        manager.requestWhenInUseAuthorization()
+        manager.delegate = self
+        manager.startUpdatingLocation()
+        
+    }
+    public func resolveLocationName(with location: CLLocation, completion: @escaping ((String?)->Void)) {
+        let geocoder = CLGeocoder()
+        geocoder.reverseGeocodeLocation(location, preferredLocale: .current) {placemarks,
+                                                                              error in
+            guard let place = placemarks?.first, error == nil else {
+                completion(nil)
+                return
+            }
+            print(place)
+            var name = ""
+            if let locality = place.locality {
+                name += locality
+            }
+            if let adminRegion = place.administrativeArea {
+                name += ",\(adminRegion)"
+            }
+            completion(name)
+        }
+    }
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        guard let location = locations.first else {
+            return
+        }
```

Comment:
it's ok, but you can shorten it like that:
```suggestion
        guard let location = locations.first else { return }
```
but it's up to you and the project's code style